packages/reputation-miner/ReputationMiner.js

```diff
@@ -123,8 +123,12 @@ class ReputationMiner {
 
     this.justificationHashes = {};
     this.reverseReputationHashLookup = {};
-    const addr = await this.colonyNetwork.getReputationMiningCycle(true, { blockTag: blockNumber });
-    const repCycle = new ethers.Contract(addr, this.repCycleContractDef.abi, this.realWallet);
+    const repCycle = await this.getActiveRepCycle(blockNumber)
+
+    if (repCycle.address === ethers.constants.AddressZero) {
+      console.log(`WARNING: No active mining cycle found for block number ${blockNumber}`);
+      return;
```

Comment:
Continuing [this](https://github.com/JoinColony/colonyNetwork/pull/631#commitcomment-33803015) discussion in context here. 
On the topic of handling control flow when getting the active mining cycle fails, looking through the instances of `getActiveRepCycle` in the miner, I think it might be more appropriate to `throw` an error instead of just `return`. In the Network there should always be an active mining cycle, except for when it is asked for from an old block before mining process has started (i.e. `blockNumber < 548534` on Goerli) which I can't see being possible with the current code, but I need to look harder later.
Was there a specific case you had in mind when you introduced the `return` here?

Reply:
Yup! The first time you call `startNextCycle`, it emits a `ReputationMiningCycleComplete` event, but there is actually no mining cycle in the previous block. So if you `sync` from block 1, or even from the block where the network was deployed (which seems like a reasonable behavior), the first block that `sync` looks at has no mining cycle. If we were to `throw` there, then the sync would stop, while the `return` functions more like a `continue`.

An alternative would be to tweak how the mining cycle emits events so `ReputationMiningCycleComplete` is emitted closer to where the previous cycle is destroyed, maybe even in `confirmNewHash` if possible.

Reply:
It may not be worth looking at this close to the finish line but glancing at `initialiseReputationMining` and `startNextCycle` it does seem like there's similar logic which could be merged / re-arranged.

Reply:
Considering the mining cycle has already been initialised on Goerli and very close to being initialised on mainnet, reengineering the initialisation of the mining process is probably a very unlikely option.
Instead perhaps we can change the default block number for syncing to be the correct one when mining was initialised, i.e. `548534` for goerli.

Reply:
Ok, I can live with that.

Reply:
Here's what I've got: https://github.com/JoinColony/colonyNetwork/pull/631/commits/f332c8d18fd75a616415ee922c4721851185e1b8

I'm certain that there is a more elegant way to handle the error messages in `index.js` than what I did, if anyone has a suggestion.